portfolio/src/main/webapp/script.js

```diff
@@ -32,8 +32,12 @@ function addRandomFact() {
 }
 
 function sayHello() {
-    fetch("/data").then(response => response.text()).then((text) => {
-    document.getElementById('hello-container').innerText = text;
+    fetch("/data").then(response => response.json()).then((text) => {
+        const greetings = document.getElementById('hello-container');
+        greetings.innerHTML = '';
+        greetings.innerHTML += text[0];
+        greetings.innerHTML += text[1];
+        greetings.innerHTML += text[2];
```

Comment:
As with my other comment, I assumed you wanted to display a single greeting. If you want to display all of them then your original change would actually be better. Sorry for the confusion. I still may not be understanding something since html/javascript are not my forte.

The reason this code isn't optimal is because you have hard-coded the number of greetings to display. If you change the number of greetings on the server-side, you will have to change the code you have here. It is often best-practice to assume that anything that can change with your inputs eventually will. Since your input is an array you want to be able to handle it changing length and having zero elements.

Another thing to keep in mind is that you want someone to be able to know what each variable is just by reading one side of the code. Here, 'text' is a little ambiguous. You could make it more clear by using the variable name textList, or by using a for loop to add each element of text to the greetings.

Reply:
The walkthrough said to hardcode 3 inputs for now just to get the whole parsing and display working. I think as I keep going, it will tell me to take the steps to make the code more functional so that the elements/array size can change as the server receives comments.

Thanks for the last note, I will go ahead and implement that and send another commit